stage_tester/locustfile.py

```diff
@@ -0,0 +1,122 @@
+from gevent import monkey; monkey.patch_socket()
+from locust import HttpUser, task, between, run_single_user, events
+import os
+import st
+import requests
+import json
+import configparser
+import random
+import logging
+
+
+conf_file = os.getenv("CONFIG_FILE_PATH", "./dev.conf")
+env_var_prefix = os.getenv("ENV_VAR_PREFIX", "LOAD_TEST_DEV_")
+
+class ApiUser(HttpUser):
+
+    wait_time = between(0.1, 1) 
+    # for debug only, will be overwritten by configuration
+    host = 'https://console.redhat.com'
+
+    def on_start(self):
+
+        self.configure()
+        self.login()
+
+
+    def configure(self):
+        cfg = configparser.ConfigParser()
+        cfg.read(conf_file)
+
+        self.offline_token = cfg['load_test']['offline_token']
+        self.client_id = cfg['load_test']['client_id']
+        self.api_url = cfg['load_test']['api_url']
+        self.login_url = cfg['load_test']['login_url']
+        self.organization = cfg['load_test']['organization']
+        
+        self.clusters = []
+                    
+
+    def login(self):
+        
+        headers = {
+            'Content-Type': 'application/x-www-form-urlencoded',
+        }
+        payload = "grant_type=refresh_token&client_id={}&refresh_token={}".format(
+            self.client_id,
+            self.offline_token
+        )
+        response = requests.post(self.login_url, headers=headers, data=payload)
+        self.access_token = response.json()['access_token']
+
+
+    @task
+    def retrieve_org_overview(self):
+
+        self.get_api_call(
+            '/api/insights-results-aggregator/v1/org_overview', self.access_token)
+
+
+    @task
+    def retrieve_gathering_conditions(self):
+
+        self.get_api_call(
+            '/api/gathering/gathering_rules', self.access_token)
+
+
+    @task
+    def retrieve_cluster_list(self):
+        
+        path = f'/api/insights-results-aggregator/v1/organizations/{self.organization}/clusters'
+        response = self.get_api_call(path, self.access_token)
+        payload = response.json()
+        if "clusters" in payload:
+            self.clusters = payload["clusters"]
+
+    @task
+    def retrieve_additional_info(self):
+        self.get_api_call(
+            '/api/insights-results-aggregator/v1/info', self.access_token)
+
+    @task
+    def retrieve_results_for_cluster(self):
+        
+        if self.clusters:
```

Comment:
Problem with this one is that it depends  on another test and it might not be executed before this one. Why not making a cached_property for the clusters?
```
@cached_property
    def clusters(self):
        path = f'/api/insights-results-aggregator/v1/organizations/{self.organization}/clusters'
        response = self.get_api_call(path, self.access_token)
        payload = response.json()
        return payload["clusters"]
```